cumulusci/robotframework/CumulusCI.py

```diff
@@ -32,23 +32,38 @@ def __init__(self, org_name=None):
         if not org_name:
             org_name = 'dev'
         self.org_name = org_name
-        self.sf = self._init_api()
-        self.tooling = self._init_api('tooling/')
         # Turn off info logging of all http requests 
         logging.getLogger('requests.packages.urllib3.connectionpool').setLevel(logging.WARN)
 
     @property
-    def config(self):
-        if not hasattr(self, '_config'):
-            self._config = self._init_config()
-        return self._config
+    def project_config(self):
+        if not hasattr(self, '_project_config'):
+            self._project_config = CliConfig().project_config
+        return self._project_config
+    
+    @property
+    def keychain(self):
+        return self.project_config.keychain
 
     @property
     def org(self):
         if not hasattr(self, '_org'):
-            self._org = self.config.keychain.get_org(self.org_name)
+            self._org = self.keychain.get_org(self.org_name)
         return self._org
 
+    @property
+    def sf(self):
+        if not hasattr(self, '_sf'):
```

Comment:
this is maybe a matter of style, but I'd prefer to see all these instance attrs set to None in `__init__` and these checks be e.g. `if self._sf is None`. if the instance attr isn't declared in __init__, ide complete, pylint, etc... struggle with it.